src/store/modules/nft.js

```diff
@@ -399,6 +399,35 @@ const actions = {
     commit(TYPES.NFT_SET_NFT_CLASS_FIAT_PRICE_INFO, { classId, data });
     return data;
   },
+  async fetchNFTClassAggregatedInfo({ commit, dispatch }, classId) {
+    const {
+      classData,
+      iscnData,
+      owners,
+      listings,
+      purchaseInfo,
+    } = await this.$api.$get(api.getNftClassMetadata(classId));
+    const iscnId = classData.parent.iscn_id_prefix;
+    commit(TYPES.NFT_SET_NFT_CLASS_METADATA, { classId, metadata: classData });
+    commit(TYPES.NFT_SET_ISCN_METADATA, { iscnId, data: iscnData });
+    commit(TYPES.NFT_SET_NFT_CLASS_OWNER_INFO, { classId, info: owners });
+    commit(TYPES.NFT_SET_NFT_CLASS_LISTING_INFO, { classId, info: listings });
+    // skip for non Writing NFT
+    if (purchaseInfo) {
+      commit(TYPES.NFT_SET_NFT_CLASS_PURCHASE_INFO, {
+        classId,
+        info: purchaseInfo,
+      });
+    }
+    if (classData.iscn_owner) {
+      const promise = dispatch(
+        'lazyGetUserInfoByAddress',
+        classData.iscn_owner
+      );
+      // Need to await if the action fires in during SSR
+      if (!process.client) await promise;
```

Comment:
Better to let the caller test for `process.client` and await, which page would have issue if we don't await these promises?

Reply:
Was copied from [here](https://github.com/likecoin/liker-land/blob/master/src/store/modules/nft.js#L528-L533).
First saw in [this commit](https://github.com/likecoin/liker-land/pull/857/commits/967e95ce00f4862f8bd8cf7bdcd4482e6e86bb10).

Reply:
@nwingt what does this break if no await?

Reply:
if not await, creator will sometime be missing in NFT Class details page
<img width="343" alt="image" src="https://github.com/likecoin/liker-land/assets/17264716/38426f6f-66d5-48db-8cb7-8dec4be8ac57">